subsys/net/ip/net_pkt.c

```diff
@@ -966,7 +966,7 @@ static size_t pkt_buffer_length(struct net_pkt *pkt,
 		max_len = MAX(max_len, NET_IPV4_MTU);
 	} else { /* family == AF_UNSPEC */
 #if defined (CONFIG_NET_L2_ETHERNET)
-		if (net_if_l2(net_pkt_iface(pkt)) ==
+		if (net_pkt_iface(pkt) && net_if_l2(net_pkt_iface(pkt)) ==
```

Comment:
why not instead doing the check always in net_if_l2()? 

That said, iface should not be null in real hw, unit tests are indeed doing tricky things something for simplification.

Reply:
Sure, we can add that check to L2.